core/src/test/scala/com/nvidia/spark/rapids/tool/qualification/QualificationSuite.scala

```diff
@@ -1373,6 +1373,58 @@ class QualificationSuite extends BaseTestSuite {
     }
   }
 
+  test("test SMJ corner case not supported on left outer join") {
+    TrampolineUtil.withTempDir { outParquetFile =>
+      TrampolineUtil.withTempDir { eventLogDir =>
+        val (eventLog, _) = ToolTestUtils.generateEventLog(eventLogDir, "smjLeftJoin") { spark =>
+          import spark.implicits._
+          val data = Seq(("A", 20, "M", "2024-01-01"),
+            ("B", 25, "M", "2024-12-12"), ("C", 30, "F", "2022-03-04"))
+              .toDF("name", "id", "gender", "dob_str")
+
+          data.write.format("parquet").mode("overwrite").save(outParquetFile.getCanonicalPath)
+          spark.read.parquet(outParquetFile.getCanonicalPath).createOrReplaceTempView("df")
```

Comment:
Can we create temp view directly from the `data` df? We could avoid writing and reading.